service/controller/app/resource/chartoperator/create_test.go

```diff
@@ -0,0 +1,414 @@
+				"3def4-cert-operator":      false,
+				"3def4-cluster-autoscaler": false,
+				"3def4-kiam":               false,
+			},
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d: %s", i, tc.name), func(t *testing.T) {
+			var err error
+
+			cr, err := key.ToApp(tc.chartoperator)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			schemeBuilder := runtime.SchemeBuilder{
+				v1alpha1.AddToScheme,
+			}
+
+			err = schemeBuilder.AddToScheme(scheme.Scheme)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			fakeLogger := microloggertest.New()
+
+			var r *Resource
+			{
+				objs := tc.apps
+				objs = append(objs, tc.chartoperator)
+
+				fakeCtrlClient := fake.NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithRuntimeObjects(objs...).
+					Build()
+
+				fakeK8sClient := clientgofake.NewSimpleClientset()
+
+				var valuesService *values.Values
+				{
+					c := values.Config{
+						K8sClient: fakeK8sClient,
+						Logger:    fakeLogger,
+					}
+
+					valuesService, err = values.New(c)
+					if err != nil {
+						t.Fatalf("error == %#v, want nil", err)
+					}
+				}
+
+				c := Config{
+					CtrlClient: fakeCtrlClient,
+					FileSystem: afero.NewMemMapFs(),
+					K8sClient:  fakeK8sClient,
+					Logger:     fakeLogger,
+
+					ChartNamespace:    "giantswarm",
+					WorkloadClusterID: "1abc2",
+					Values:            valuesService,
+				}
+
+				r, err = New(c)
+				if err != nil {
+					t.Fatalf("error == %#v, want nil", err)
+				}
+			}
+
+			var ctx context.Context
+			{
+				fakeCtrlClient := fake.NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithRuntimeObjects(tc.charts...).
+					Build()
+
+				fakeClient := k8sclienttest.NewClients(k8sclienttest.ClientsConfig{
+					CtrlClient: fakeCtrlClient,
+					K8sClient:  clientgofake.NewSimpleClientset(),
+				})
+
+				defaultContent := helmclient.ReleaseContent{
+					Status: helmclient.StatusDeployed,
+				}
+
+				c := controllercontext.Context{
+					Clients: controllercontext.Clients{
+						K8s: fakeClient,
+						Helm: helmclienttest.New(helmclienttest.Config{
+							DefaultReleaseContent: &defaultContent,
+						}),
+					},
+				}
+
+				ctx = controllercontext.NewContext(context.Background(), c)
+			}
+
+			err = r.EnsureCreated(ctx, &cr)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			var appList v1alpha1.AppList
+			err = r.ctrlClient.List(ctx, &appList)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			for _, a := range appList.Items {
+				_, ok := a.GetAnnotations()[AppOperatorTriggerReconciliation]
+				expected := tc.expectAnnotation[a.ObjectMeta.Name]
+				if expected != ok {
+					t.Fatalf("%s: expected %t, got %t", a.ObjectMeta.Name, expected, ok)
+				}
+			}
+		})
+	}
+}
+
+func Test_Resource_triggerReconciliation(t *testing.T) {
+	tests := []struct {
+		name             string
+		apps             []runtime.Object
```

Comment:
Using `runtime.Object` is a bit strange here. Could we use the actual types and then convert them when setting up the fake client?